app/geo/gui/src/ts/twxplore/gui/geo/components/map/Map.tsx

```diff
@@ -126,6 +128,44 @@ const MapImpl: React.FunctionComponent = () => {
             options: {centerMap: true, readOnly: true},
           })
         );
+
+        /*
+        A lot of hardcodeding here. I could probably iterate through the properties later on.
+        */
+        const typeRanges = Object.assign({}, state.typesRanges);
+        for (const feature of featuresInState) {
+          const typeRange = typeRanges[String(feature.type)];
+          if (typeRange) {
+            if (feature.timestamp) {
+              if (feature.timestamp < typeRange.timestamp.min!)
+                typeRange.timestamp.min = feature.timestamp;
+              else if (feature.timestamp > typeRange.timestamp.max!)
+                typeRange.timestamp.max = feature.timestamp;
+            }
+
+            if (feature.frequency) {
+              if (feature.frequency < typeRange.frequency.min!)
+                typeRange.frequency.min = feature.frequency;
+              else if (feature.frequency > typeRange.frequency.max!)
+                typeRange.frequency.max = feature.frequency;
+            }
+          } else {
+            //this is first time coming across type. Add a filter for it and create a typeRange object for it.
+            addFilter(feature.type);
```

Comment:
if (!typeRange from state) { addFilter(feature.type); }
Then intercept the Kepler action in our reducer instead of creating the type range here.

Reply:
That would mean you don't need the replaceTypeRanges action, I think?